src/Microsoft.ML.PipelineInference/AutoMlUtils.cs

```diff
@@ -15,21 +15,41 @@ namespace Microsoft.ML.Runtime.PipelineInference
 {
     public static class AutoMlUtils
     {
-        public static AutoInference.RunSummary ExtractRunSummary(IHostEnvironment env, IDataView data, string metricColumnName)
+        public static AutoInference.RunSummary ExtractRunSummary(IHostEnvironment env, IDataView result, string metricColumnName, IDataView trainResult = null)
         {
-            double metricValue = 0;
+            double testingMetricValue = 0;
+            double trainingMetricValue = -1d;
             int numRows = 0;
-            var schema = data.Schema;
-            schema.TryGetColumnIndex(metricColumnName, out var metricCol);
+            var schema = result.Schema;
+            bool hasIndex = schema.TryGetColumnIndex(metricColumnName, out var metricCol);
+            env.Check(hasIndex, $"Schema does not contain metric column: {metricColumnName}");
 
-            using (var cursor = data.GetRowCursor(col => col == metricCol))
+            using (var cursor = result.GetRowCursor(col => col == metricCol))
             {
                 var getter = cursor.GetGetter<double>(metricCol);
-                cursor.MoveNext();
-                getter(ref metricValue);
+                bool moved = cursor.MoveNext();
+                env.Check(moved, "Could not move cursor forward. (Results dataset has no rows to extract.)");
+                getter(ref testingMetricValue);
+                env.Check(!cursor.MoveNext(), "Encountered additonal, unexpected results row.");
             }
 
-            return new AutoInference.RunSummary(metricValue, numRows, 0);
+            if (trainResult != null)
+            {
+                var trainSchema = trainResult.Schema;
+                env.Check(trainSchema.TryGetColumnIndex(metricColumnName, out var trainingMetricCol),
+                    $"Schema does not contain metric column: {metricColumnName}");
```

Comment:
>$"Schema does not contain metric column: {metricColumnName}" [](start = 20, length = 60)

For anything involving *creating* a string rather than a string literal we'd prefer not to put that in the check message... that is, rephrasing it as an `if` followed by a `throw env.Except` would be helpful.

Also note: since this about `metricColumnName`, might be nice for it to be an `env.ExceptParam`.